src/protocols/types/SurfaceState.hpp

```diff
@@ -3,10 +3,30 @@
 #include "../../helpers/math/Math.hpp"
 #include "../WaylandProtocol.hpp"
 #include "./Buffer.hpp"
+#include "helpers/time/Time.hpp"
 
 class CTexture;
 class CDRMSyncPointState;
 
+struct SSurfaceStateFrameCB {
```

Comment:
do not move the callback resource class like this. Keep it as it was, you can add methods to it. Resources need to be uniform, this is inviting issues down the line.

There's nothing you added here that can't just be added to CWLCallbackResource

you can keep WP<> refs in the surface state